src/contexts/AzureMapDataSourceContext.tsx

```diff
@@ -46,6 +48,27 @@ const AzureMapDataSourceStatefulProvider = ({
         dref.add(collection)
       }
     }
+    return () => {
+      for (const eventType in events || {}) {
+        mref.events.remove(eventType as any, dref, events[eventType])
+      }
+
+      const getLayersDependingOnDatasource = (dst: DataSourceType) => {
+        return mref.layers.getLayers().filter((l) => {
+          if ((l as atlas.layer.SymbolLayer).getSource) {
+            const sourceLayer = (l as atlas.layer.SymbolLayer).getSource()
+            const dsId = typeof sourceLayer === 'string' ? sourceLayer : sourceLayer.getId()
+            return dsId === dst.getId()
+          }
+          return false
+        })
+      }
+
+      getLayersDependingOnDatasource(dref).forEach((l) => {
+        mref.layers.remove(l.getId() ? l.getId() : l)
+      })
+      mref.sources.remove(dref)
```

Comment:
Thanks @rvdkooy, this is great! 

Do you mind apply the same fix to `AzureMapVectorTileSourceProvider.tsx` as well?  It can be tested with react-azure-maps-playground under http://localhost:3000/custom-traffic

Reply:
sorry, wasn't able to have a look at it for a while.
I also fixed it in `AzureMapVectorTileSourceProvider`

Gr,
R